### PR TITLE
test: Reference `git_ref` in action calls to allow testing updates to actions

### PR DESCRIPTION
**Pull request by jonathan-innis**
Fixes # 

**Description**

- Removes separate `actions/checkout` calls for the Action and the Repo and favors checking out the `inputs.git_ref` a single time
- Remove the `git_repo` parameter since that parameter will reference whichever repository that the action context is running inside of

**How was this change tested?**

* Manual runs on fork

**Does this change impact docs?**
- [ ] Yes, PR includes docs updates
- [ ] Yes, issue opened: # 
- [x] No

**Release Note**

```release-note

```

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/e2e-matrix.yaml [hunk 1]
```diff
@@ -55,7 +56,6 @@ jobs:
     if: (github.repository == 'aws/karpenter' && (github.event_name != 'workflow_run' || github.event.workflow_run.conclusion == 'success')) || github.event_name == 'workflow_dispatch'
     uses: ./.github/workflows/e2e-upgrade.yaml
     with:
-      git_repo: aws/karpenter
       from_git_ref: v0.26.1
```
Comment (bwagner5) [inline .github/workflows/e2e-matrix.yaml:59]: do we need to set this to v0.27.5?

Comment (jonathan-innis) [reply .github/workflows/e2e-matrix.yaml:59]: Good callout

Comment (jonathan-innis) [reply .github/workflows/e2e-matrix.yaml]: Updated 👍

**Review by bwagner5 — APPROVED** [review]
lgtm

**Pull request merged**